agent/util.go

```diff
@@ -116,9 +116,14 @@ func setFilePermissions(path string, p FilePermissions) error {
 		if u, err := user.Lookup(p.User()); err == nil {
 			uid, _ = strconv.Atoi(u.Uid)
 			goto GROUP
+		} else {
+			switch err.Error() {
```

Comment:
Hi @jonhadfield I'm thinking the simplest thing here might just to do something like:

`return fmt.Errorf("failed to look up user %q: %v", p.User(), err)`

That way we avoid having to look at the error and always make sure whatever it is becomes visible.

Reply:
Hi @slackpad Yes, makes sense. Since I submitted the PR, go 1.9 added support for user lookup without cgo enabled, and I can see the this code has already been updated to use it.
I don't think there's any need for this PR any more, the only thing I would suggest is replacing https://github.com/hashicorp/consul/blob/144e6e7d3176138fadd69e07e6641f2164b0e484/agent/util.go#L69 
With a less specific error, as you suggested:

    return fmt.Errorf("failed to look up user %s: %v", u, err)

The current message, like the previous one, isn't very helpful, and potentially hides other scenarios, e.g. running out of file descriptors so this fails: https://golang.org/src/os/user/lookup_unix.go?h=lookupUser#L181

Reply:
Ah glad it's not cgo now - I'll go ahead and commit that new error message and close this out!